src/screens/LocalCommunity/hooks/useLocalCommunity.js

```diff
@@ -0,0 +1,172 @@
+import {debounce} from 'lodash';
+import {showMessage} from 'react-native-flash-message';
+
+import {post} from '../../../api/server';
+import {setLocalCommunity} from '../../../context/actions/localCommunity';
+import {Context} from '../../../context';
+import {Analytics} from '../../../libraries/analytics/firebaseAnalytics';
+
+const useLocalCommunity = () => {
+  const navigation = useNavigation();
+  const [search, setSearch] = React.useState('');
+  const [isVisibleSecondLocation, setIsVisibleSecondLocation] = React.useState(false);
+  const [isVisibleFirstLocation, setIsVisibleFirstLocation] = React.useState(false);
+  const [locationPost, setLocationPost] = React.useState([]);
+  const [location, setLocation] = React.useState([]);
+  const [optionsSearch, setOptionsSearch] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [locationLog, setLocationLog] = React.useState([]);
+  const [, dispatch] = React.useContext(Context).localCommunity;
+
+  const handleKeyExtractor = (item, index) => {
+    return index.toString();
+  };
+
+  const onPressSecondLocation = (status) => {
+    setIsVisibleSecondLocation(status);
+    setSearch('');
+  };
+
+  const onPressFirstLocation = (status) => {
+    setIsVisibleFirstLocation(status);
+    setSearch('');
+  };
+
+  const onBack = () => {
+    navigation.goBack();
+  };
+
+  const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1);
+
+  const handleSearch = (value) => {
+    const params = {
+      name: value
+    };
+
+    post({url: '/location/list_v2', params})
+      .then((res) => {
+        setIsLoading(false);
+        if (res.status === 200) {
+          setOptionsSearch(res.data.body);
+        }
+      })
+      .catch(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const doOnLocationSearchTextDebounce = React.useCallback(debounce(handleSearch, 500), []);
+
+  const onChangeLocationSearchText = (text) => {
+    if (text.length >= 3) {
+      doOnLocationSearchTextDebounce(text);
+      setOptionsSearch([]);
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+      doOnLocationSearchTextDebounce.cancel();
+      setOptionsSearch([]);
+    }
+
+    setSearch(text);
+  };
+
+  const handleSelectedSearch = async (val, index) => {
+    const tempLocation = [...location];
+    if (tempLocation.length <= 1) {
+      tempLocation.push(val);
+    } else {
+      tempLocation[index] = val;
+    }
+    setSearch(capitalizeFirstLetter(val.neighborhood));
+    setOptionsSearch([]);
+    const locLog = [];
+    const returnTempLocation = tempLocation.map((item) => {
+      locLog.push({
+        location: `${item.city}, ${item.zip}`,
+        location_level: item.location_level
+      });
+      return item.location_id;
+    });
+    await setLocation(tempLocation);
+    await setLocationPost(returnTempLocation);
+    await setLocationLog(locLog);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleSelectedSearch` function has several `await` statements that are not necessary. The `setState` function in React is asynchronous, but it does not return a promise, so there's no need to use `await`. This could potentially slow down your application as JavaScript will wait for these lines to execute before moving on.
```diff
- await setLocation(tempLocation);
- await setLocationPost(returnTempLocation);
- await setLocationLog(locLog);
+ setLocation(tempLocation);
+ setLocationPost(returnTempLocation);
+ setLocationLog(locLog);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->